orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -269,3 +281,39 @@ def named_modules(self, *args, **kwargs):
         """Override :meth:`~torch.nn.Module.named_modules`"""
 
         yield from self._torch_module.named_modules(*args, **kwargs)
+
+    def __getattr__(self, name: str):
+        if '_torch_module' in self.__dict__:
```

Comment:
is there another way that doesn't depend on string matching?
if the variable is renamed or moved after a refactoring, things can go wrong. Maybe checking for `isinstance(self, ORTModule)`

Reply:
No, this ```if``` is checking if ```_torch_module``` attribute has already been instantiated. For example, in the ```__init__``` method of ```ORTModule``` we could have some checks or conditions that are trying to get an attribute before ```_torch_module``` is defined. So, if ```_torch_module``` is not instantiated, there is no point in forwarding the call to the user's model (since we don't have a reference to the user's model without ```_torch_module```). That is what this ```if``` condition is checking for.
I am not aware of any other way of doing this check other than checking for the attribute to exist in the ```self.__dict__```.